build/ical.js

```diff
@@ -6515,7 +6515,14 @@ ICAL.Event = (function() {
     },
 
     get endDate() {
-      return this._firstProp('dtend');
+      var end_date = this._firstProp('dtend');
+      if(end_date === null) {
```

Comment:
What the original code was also missing is to handle the case where there is a DTSTART and a DURATION but no DTEND. Aside from changing the endDate getter, handling that case would also require changing the setter for endDate. The getter for duration should be fine because changing endDate would account for that, we can optimize this for performance later if its an issue.